conformance/lang/expressions/is-expr/is_expr.balt

```diff
@@ -0,0 +1,2094 @@
+    io:println(b is map<any>); // @output false
+    io:println(b is any); // @output true
+    io:println(b is map<string>); // @output false
+
+    table<record {|int a; string b;|}> c = table [{a: 1, b: "A"}];
+    io:println(c is table<record {|readonly int a; string b;|}>); // @output false
+    io:println(c is table<map<int|string>>); // @output true
+    io:println(c is anydata); // @output true
+}
+
+Test-Case: output
+Description: Test field access expression as expression in type test expression.
+Labels: boolean, field-access-expr, float, int, is-expr, mapping-constructor-expr, module-type-defn, optional-type,
+        record-type, string, union-type
+
+type Type string|xml|boolean;
+
+type Record1 record {|
+    int|string a;
+    float? b;
+    int c?;
+    boolean d = true;
+    Type? e;
+    string...;
+|};
+
+type Record2 record {
+    int|string a;
+    float? b;
+    int c?;
+    boolean d = true;
+};
+
+function init() {
+    Record1 rec1 = {a: 1, b: (), "e": "A", "f": "B"};
+
+    io:println(rec1.a is int); // @output true
+    io:println(rec1.b is float); // @output false
+    io:println(rec1.c is int); // @output false
+    io:println(rec1.d is Type); // @output true
+    io:println(rec1.e is string); // @output true
+
+    Record2 rec2 = {a: 1, b: 2f, c: 1, d: false, "e": "A"};
+
+    io:println(rec2.a is int); // @output true
+    io:println(rec2.b is float); // @output true
+    io:println(rec2.c is int); // @output true
+    io:println(rec2.d is boolean); // @output true
+}
+
+Test-Case: output
+Description: Test optional field access expression as expression in type test expression.
+Labels: boolean, float, int, is-expr, mapping-constructor-expr, module-type-defn, optional-field-access-expr,
+        optional-type, record-type, string, union-type
+
+type Type string|xml|boolean;
+
+type Record1 record {|
+    int|string a;
+    float? b;
+    int c?;
+    boolean d = true;
+    Type? e;
+    string...;
+|};
+
+type Record2 record {
+    int|string a;
+    float? b;
+    int c?;
+    boolean d = true;
+};
+
+function init() {
+    Record1 rec1 = {a: 1, b: (), "e": "A", "f": "B"};
+
+    io:println(rec1?.a is int); // @output true
+    io:println(rec1?.b is float); // @output false
+    io:println(rec1?.c is int); // @output false
+    io:println(rec1?.d is Type); // @output true
+    io:println(rec1?.e is string); // @output true
+
+    Record2 rec2 = {a: 1, b: 2f, c: 1, d: false, "e": "A"};
+
+    io:println(rec2?.a is int); // @output true
+    io:println(rec2?.b is float); // @output true
+    io:println(rec2?.c is int); // @output true
+    io:println(rec2?.d is boolean); // @output true
+}
+
+Test-Case: output
+Description: Test xml attribute access expression as expression in type test expression.
+Labels: error-type, is-expr, module-type-defn, nil-type, optional-type, string, union-type,
+        xml-optional-attribute-access-expr, xml-required-attribute-access-expr, xmlns-decl
```

Comment:
Not sure if we need to have two labels for `module-xmlns-decl` and `xmlns-decl-stmt`